action/protocol/staking/candidate_statereader.go

```diff
@@ -478,11 +478,20 @@ func (c *candSR) readStateCandidateByName(ctx context.Context, req *iotexapi.Rea
 }
 
 func (c *candSR) readStateCandidateByAddress(ctx context.Context, req *iotexapi.ReadStakingDataRequest_CandidateByAddress) (*iotextypes.CandidateV2, uint64, error) {
-	owner, err := address.FromString(req.GetOwnerAddr())
-	if err != nil {
-		return nil, 0, err
+	var cand *Candidate
+	if len(req.GetOwnerAddr()) > 0 {
+		owner, err := address.FromString(req.GetOwnerAddr())
+		if err != nil {
+			return nil, 0, err
+		}
+		cand = c.GetCandidateByOwner(owner)
+	} else {
+		id, err := address.FromString(req.GetId())
+		if err != nil {
+			return nil, 0, err
+		}
+		cand = c.GetByIdentifier(id)
```

Comment:
is it possible `Id` and `OwnerAddress` both exist? if yes, which one takes precedence?

Reply:
updated, now if both conditions are set, it means that they need to be fulfilled at mean while.